pkgs/development/compilers/sbcl/common.nix

```diff
@@ -21,7 +22,8 @@ stdenv.mkDerivation rec {
     inherit sha256;
   };
 
-  buildInputs = [texinfo];
+  nativeBuildInputs = [ texinfo ];
+  buildInputs = lib.optionals coreCompression [ zstd ];
```

Comment:
In principle we could also support it for < 2.2.6 using `zlib` (although maybe not turn on by default).